### PR TITLE
Update target and compile Android sdk versions to v29

### PR DESCRIPTION
**Pull request by malinajirka**
Do NOT merge this PR - we plan to merge it together with PRs from gutenberg-mobile and WPAndroid.

## Description
This PR updates:
- compile and target Android SDK versions
- Android builds tools version

This change is required so we can update WPAndroid - [draft PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12947).

## How has this been tested?
I have tested the changes in WPAndroid - successful build should be enough.

## Types of changes
Updates build setup for Android

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java
Comment (mchowning) [inline packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:379]: Looks like [LineBreaker](https://developer.android.com/reference/android/graphics/text/LineBreaker) was introduced in Q. Should we do something like this?

```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
  // use LineBreaker.JUSTIFICATION_MODE...;
} else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
    // use LAYOUT.JUSTIFICATION_MODE...;
}
```

Comment (malinajirka) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:379]: Good question ;). I believe we don't need to do that. Constants are replaced during compilation and they haven't changed the values of the constants. They just moved them to a different file. Our compileSdkVersion is set to 29 so using just the constant for api 29 seems legit. Either case, I'll try to run this code on API 27 emulator to make sure I'm not wrong :).

Comment (malinajirka) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:379]: Hmm, I'm looking into this more. It seems that lint on CI complains in WPAndroid - it doesn't complain locally.

Comment (malinajirka) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java]: I think lint is quite confused, when I wrap the call it fails with:
```
  /Users/jirimalina/Code/Automattic/WordPress-Android-submodule/libs/gutenberg-mobile/gutenberg/packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:375: Error: Unnecessary; SDK_INT is never < 21 [ObsoleteSdkInt]
                  if (Build.VERSION.SDK_INT >= VERSION_CODES.Q) {
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

I think the original code would work just fine for the reasons stated above. However, I wrapped the code with the "build version" ifs. It's not nice but hopefully both local and CI lint will be happy :P.

Comment (mchowning) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java]: > I think lint is quite confused



Nice job figuring out the magic incantation to make lint happy! 😄 

It looks like we're missing an import for the `@Require` annotation, which is causing the build failure. Also, what would you think about adding a comment explaining that we're jumping through these weird hoops because of (what looks like) a bug in lint? Alternatively, with a comment like that, I'd be fine just suppressing the relevant lint warning if you preferred to do that. I'm fine with either approach.

Comment (mchowning) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java]: I have filed an issue (https://github.com/wordpress-mobile/gutenberg-mobile/issues/2653) for us to fix this "code change in gutenberg causes no problems until we try to run lint in WPAndroid" problem. Wanted to let you know because I figured you might have some thoughts on that given your recent experience. 😉

**Review by mchowning — APPROVED** [review]
Looks good and tested this out and everything is working well. I'll wait for your go-ahead to start the merge train @malinajirka (or feel free to ping someone else if I'm not around).

**Pull request merged**
